### ISSUE TITLE
Crossterm panics when setting theme

### ISSUE THREAD

**Issue by chrisvest** [gyscos/cursive#352]
# Calling set_theme with the crossterm backend hits a code path that is unimplemented.

Specifically, this place: https://github.com/gyscos/Cursive/blob/master/src/backend/crossterm.rs#L293

I don't know if this was forgotten about in the original PR or intentionally punted. XV always sets a theme when it starts, so it always panics on startup when built with the crossterm backend.

```
Panic in xv 0.1.1, 2019-05-26T16:14:19+02:00.
System: linux (unix), x86_64.
Location: /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/cursive-0.12.0/src/backend/crossterm.rs:293:13.

stack backtrace:
   0: backtrace::backtrace::libunwind::trace::h8ddc7da5b84baa88 (0x560d2585d04c)
             at /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.16/src/backtrace/libunwind.rs:80
      backtrace::backtrace::trace_unsynchronized::h382fc165cd29745b
             at /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.16/src/backtrace/mod.rs:55
   1: backtrace::backtrace::trace::hf72464fdbee526f8 (0x560d2585cfcd)
             at /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.16/src/backtrace/mod.rs:46
   2: backtrace::capture::Backtrace::create::h720ed5cf10bf8045 (0x560d25854684)
             at /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.16/src/capture.rs:132
   3: backtrace::capture::Backtrace::new::h355a6f3c77f26977 (0x560d25854573)
             at /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.16/src/capture.rs:97
   4: xv::panic_hook::report_crash::h77e3b591e63723b0 (0x560d2583279e)
             at src/panic_hook.rs:58
   5: xv::panic_hook::install::{{closure}}::h7a3fe381ae0cd7ae (0x560d257772fa)
             at src/panic_hook.rs:19
   6: rust_panic_with_hook (0x560d259ac288)
             at src/libstd/panicking.rs:478
   7: continue_panic_fmt (0x560d259abd21)
             at src/libstd/panicking.rs:381
   8: begin_panic_fmt (0x560d259abc6e)
             at src/libstd/panicking.rs:336
   9: cursive::backend::crossterm::with_color::hc37b2319c0c6e18e (0x560d2592770a)
             at /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/cursive-0.12.0/src/backend/crossterm.rs:293
  10: cursive::backend::crossterm::Backend::apply_colors::h0baa01a9d2f07994 (0x560d25924464)
             at /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/cursive-0.12.0/src/backend/crossterm.rs:58
  11: <cursive::backend::crossterm::Backend as cursive::backend::Backend>::clear::hd8c9df8f8b73a5d7 (0x560d25926243)
             at /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/cursive-0.12.0/src/backend/crossterm.rs:212
  12: cursive::cursive::Cursive::clear::h5f22713a5ff4f380 (0x560d25909ede)
             at /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/cursive-0.12.0/src/cursive.rs:423
  13: cursive::cursive::Cursive::set_theme::hda12235599af9f90 (0x560d25909e59)
             at /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/cursive-0.12.0/src/cursive.rs:416
  14: xv::xv_tui::run_tui::ha44a11d7657d893d (0x560d25829734)
             at src/xv_tui.rs:23
  15: xv::main::h7d1c50c5179af321 (0x560d2577ea7b)
             at src/main.rs:70
  16: std::rt::lang_start::{{closure}}::h0fe33f3236dbaac2 (0x560d2582bed1)
             at /rustc/3c235d5600393dfe6c36eeed34042efad8d4f26e/src/libstd/rt.rs:64
  17: {{closure}} (0x560d259abba2)
             at src/libstd/rt.rs:49
      do_call<closure,i32>
             at src/libstd/panicking.rs:293
  18: __rust_maybe_catch_panic (0x560d259af1a9)
             at src/libpanic_unwind/lib.rs:87
  19: try<i32,closure> (0x560d259ac6ac)
             at src/libstd/panicking.rs:272
      catch_unwind<closure,i32>
             at src/libstd/panic.rs:388
      lang_start_internal
             at src/libstd/rt.rs:48
  20: std::rt::lang_start::hd4760abe60d8dbfa (0x560d2582be98)
             at /rustc/3c235d5600393dfe6c36eeed34042efad8d4f26e/src/libstd/rt.rs:64
  21: main (0x560d2577eb69)
  22: __libc_start_main (0x7fdd3beb1412)
  23: _start (0x560d2576b20d)
  24: <unknown> (0x0)
```

# Environment

* Cursive version 0.12.0 from crates.io.

**Comment by chrisvest** [comment]
Ping @TimonPost

**Comment by gyscos** [comment]
The problem seems to be the `TerminalDefault` color not being currently implemented.
Should be easy to patch; in the meantime, avoiding this "color" (or this backend) may help.

**Comment by chrisvest** [comment]
Looks like this has been fixed as of at least 0.14.
